conanfile.py

```diff
@@ -229,6 +231,14 @@ def requirements(self):
             self.requires("czmq/4.2.0")
             self.options['czmq'].shared = True
 
+            #If czmq is needed, disable crypto for libzip to prevent openssl conflict:
+            # 'czmq/4.2.0' requires 'openssl/1.1.1m' while 'libcurl/7.86.0' requires 'openssl/1.1.1s'
+            self.options['libzip'].crypto = False
```

Comment:
It's not necessary to disable libzip. We can override openssl requirement in this recipe. Downstream can always override upstream requirement.

Reply:
Actually you can do this on the conan command line using `--require-override` to specify  openssl version.
If the version you specify differs from czmq or libcurl's requirement, they will be rebuilt with a different package id.
So there's nothing to worry about.

Reply:
The point is that we should specify minimal requirements to the upstream, and leave the ultimate power of opting-out/overriding to our downstream.